packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -612,17 +724,24 @@ export class AccountHandler {
   async finishSetup(request: AuthRequest) {
     this.log.begin('Account.finishSetup', request);
     const form = request.payload as any;
-    const authPW = form.authPW;
+    const { authPW, authPWVersion2, wrapKb, wrapKbVersion2, clientSalt, token } = form;
     let uid;
     try {
-      const payload = (await jwt.verify(form.token, {
+      const payload = (await jwt.verify(token, {
         typ: 'fin+JWT',
         ignoreExpiration: true,
       })) as any;
       uid = payload.uid;
       form.uid = payload.uid;
       const account = await this.db.account(uid);
-      await this.setPasswordOnStubAccount(account, authPW);
+      await this.setPasswordOnStubAccount({
+        account,
+        authPW,
+        authPWVersion2,
+        wrapKb,
+        wrapKbVersion2,
+        clientSalt,
+      });
```

Comment:
Shouldn't we set the wrapWrapKb2 value based on the the password derived values here?  warpWrapKb2 was randomly generated when the stub account was created so it's not going to unwrap to the same kB as v1.

Reply:
The client suplies wrapKb2, and we then derive wrapWrapKb2 from this value. The important thing is that that wrapKb2 is derived from the existing kB. This has to happenn client side, since only the client can determine kB.

Reply:
> The client suplies wrapKb2, and we then derive wrapWrapKb2 from this value.
Exactly.  But where is that happening?  Looking at the patch, wrapWrapKb2 is still the value generated on https://github.com/mozilla/fxa/pull/16249/files#diff-964155b6fe5094c1fdb982eca65dcf044c10cbbf89d3b8878bc596138b35c108R28 after this.

Reply:
Ah I see. It appears this handler didn't have any integration tests coverage (ie test/remote). Let me try and add something there... At any rate, once the test is created, the client must supply and overwrite the wrapWrapKb2